pkg/clustermanager/cluster_manager.go

```diff
@@ -252,9 +256,40 @@ func (c *ClusterManager) generateWorkloadKubeconfig(ctx context.Context, cluster
 	return writtenFile, nil
 }
 
-func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, clusterToDelete *types.Cluster) error {
+func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, clusterToDelete *types.Cluster, provider providers.Provider, clusterSpec *cluster.Spec) error {
 	return c.Retrier.Retry(
 		func() error {
+			if clusterSpec.IsManaged() {
```

Comment:
i think we need to retry each kubectl call instead of all together

Reply:
Added  "--ignore-not-found=true" to delete command making them idempotent.